packages/insured-bridge-relayer/src/CrossDomainFinalizer.ts

```diff
@@ -44,8 +44,11 @@ export class CrossDomainFinalizer {
   async checkForBridgeableL2TokensAndBridge() {
     this.logger.debug({ at: "CrossDomainFinalizer", message: "Checking bridgeable L2 tokens" });
 
-    // Fetch all whitelisted tokens on the particular l2 chainId.
-    const whitelistedL2Tokens = this.l1Client.getWhitelistedL2TokensForChainId(this.l2Client.chainId.toString());
+    // Fetch all whitelisted tokens on the particular l2 chainId. Remove the DAI Optimism address from the whitelist as
+    // we don't want to finalize DAI actions due to this not working over the canonical Optimism bridge.
+    const whitelistedL2Tokens = this.l1Client
+      .getWhitelistedL2TokensForChainId(this.l2Client.chainId.toString())
+      .filter((address) => address !== "0xDA10009cBd5D07dd0CeCc66161FC93D7c9000da1"); // Remove DAI on Optimism address.
```

Comment:
Can you please just add L2_TOKEN_BLACKLIST to the RelayerConfig and filter more upstream from the L1Client? Would be a lot cleaner and easier to change down the line

Reply:
we could do this...but is it worth it? we are going to be deprecating this bot within a short period of time and I don't really think it matters to have some hot glue holding some of it together until we deprecate this. you ok with this?

Reply:
yea lgtm